packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol

```diff
@@ -0,0 +1,2923 @@
+                    .bids[currentLatestRefundedBidId]
+                    .prevBidId;
+            }
+
+            // if scrolled off end of list, then find next slot with bids
+            if (currentLatestRefundedBidId == 0) {
+                // past head of current slot's linked list, so need to find next
+                // bid slot with bids
+                // @dev not possible to not find next slot during auto-refund,
+                // so no need to handle case where slot not found
+                currentLatestRefundedBidSlotIndex = _getMinSlotWithBid({
+                    RAMProjectConfig_: RAMProjectConfig_,
+                    startSlotIndex: uint16(
+                        currentLatestRefundedBidSlotIndex + 1
+                    )
+                });
+                // current bid is now the tail of the linked list
+                currentLatestRefundedBidId = RAMProjectConfig_.tailBidIdBySlot[
+                    currentLatestRefundedBidSlotIndex
+                ];
+            }
+
+            // @dev minter is in State C, so bid must not have been minted or
+            // refunded due to scrolling logic of admin mint and refund
+            // functions available for use while in State C. The bid may have
+            // been previously settled, however.
+
+            // Settle & Refund the Bid
+            _settleAndRefundBid({
+                projectId: projectId,
+                coreContract: coreContract,
+                projectPrice: projectPrice,
+                slotIndex: uint16(currentLatestRefundedBidSlotIndex),
+                bidId: currentLatestRefundedBidId,
+                minterRefundGasLimit: minterRefundGasLimit
+            });
+            // increment loop counter and current num bids refunded
+            unchecked {
+                ++numRefundsIssued;
+            }
+        }
+
+        // finally, update auction metadata storage state from memoized values
+        // @dev safe to cast currentNumBidsErrorRefunded to uint24
+        RAMProjectConfig_.numBidsErrorRefunded += uint24(numRefundsIssued);
+        // @dev safe to cast to uint32 because directly derived from bid ID
+        RAMProjectConfig_.latestRefundedBidId = uint32(
+            currentLatestRefundedBidId
+        );
+    }
+
+    /**
+     * @notice Function to settle and refund bids to resolve E1 error state.
+     * @param projectId Project ID to refund bid for.
+     * @param coreContract Core contract address for the given project.
+     * @param projectPrice Price of a token for the given project.
+     * @param slotIndex Slot index of bid.
+     * @param bidId ID of bid to settle.
+     * @param minterRefundGasLimit Gas limit to use when refunding bidder
+     */
+    function _settleAndRefundBid(
+        uint256 projectId,
+        address coreContract,
+        uint256 projectPrice,
+        uint256 slotIndex,
+        uint256 bidId,
+        uint256 minterRefundGasLimit
+    ) internal {
```

Comment:
should be private + follow order of functions

Reply:
Updated!